rfc/tableAvailabilityMarkdown.md

```diff
@@ -0,0 +1,108 @@
+# Table resource supporting filtered availability
+
+This RFC implements [Github Issue 306](https://github.com/yahoo/fili/issues/306)
+
+## Goals
+
+1. Show the queryable range of time as part of the Logical Table endpoint view.
+2. Allow the queryable range to reflect constraints as would be found in a data request.
+  * Required metrics and dimensions
+  * Applied filters
+
+
+## Calculating availability
+
+Availability for a logical table shall be defined as the maximal set of intervals that the table may be able to respond
+ to.  Or put another way, the table can certainly not respond to any intervals beyond the range marked available.  As
+ such the calculation will involve unioning the available intervals for all physical tables backing the logical table.
+ 
+ Constrained availability will be defined as the availability for a table given a set of query constraints.  This will
+ be the result of first filtering out physical tables on the logical table which cannot answer the query.  This may end
+ up excluding ALL physical tables, meaning an empty availability.
+ 
+ Existing code for constrained physical tables should be able to be repurposed:
+
+1. Build a `QueryPlanningConstraint` from the TableApiRequest
+    From: `DruidQueryBuilder`
+    ```java 
+        TableGroup group = logicalTable.getTableGroup();
+    
+        // Resolve the table from the the group, the combined dimensions in request, and template time grain
+        QueryPlanningConstraint constraint = new QueryPlanningConstraint(request, template);
+    
+    ```
+    * In the case of unconstrained queries, ensure that an 'unconstrained' constraint is built.
```

Comment:
By unconstrainted I mean that it's constrained only by fields specified in the ApiRequest.  So if the ApiRequest doesn't list dimensions, no dimensional constraints are applied, and so forth.